firedrake/mesh.py

```diff
@@ -149,15 +150,19 @@ def facet_cell_map(self):
                        "facet_to_cell_map")
 
 
-def _from_gmsh(filename):
-    """Read a Gmsh .msh file from `filename`"""
+def _from_gmsh(filename, comm=None):
+    """Read a Gmsh .msh file from `filename`.
 
+    :kwarg comm: Optional communicator to build the mesh on (defaults to
+        COMM_WORLD).
+    """
+    comm = comm or COMM_WORLD
```

Comment:
Use library comm_world (not user) although petsc dups the comm, so probably fine.